### PR TITLE
add each label as a separate PropertySource

### PR DESCRIPTION
**Pull request by sophiaso**
## Description
Add label name to the PropertySource, as the PropertySource should be independent between different labels.

### CODE REVIEW

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigProperties.java [hunk 1]
```diff
@@ -232,4 +238,27 @@ public void validateAndInit() {
...
+    /**
+     * @return List of reversed label values, which are split by the separator, the latter label has higher priority
+     */
+    public List<String> getLabels() {
+        if (!StringUtils.hasText(this.getLabel())) {
+            return EMPTY_LABEL_ONLY;
+        }
+
+        List<String> labels =  Arrays.stream(this.getLabel().split(LABEL_SEPARATOR))
+                .filter(label -> StringUtils.hasText(label))
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigProperties.java]: `StringUtils::hasText` and `String::trim`?

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigWatch.java [hunk 2]
```diff
@@ -91,40 +88,44 @@ public void watchConfigKeyValues() {
...
 
-    private boolean mapInclude(Map<String, String> map, String key, String value) {
-        return map.containsKey(key) && (map.get(key) != null && map.get(key).equals(value));
+    private boolean isEtagChanged(Set<String> newEtagSet, Set<String> prevEtagSet) {
+        if (newEtagSet.size() != prevEtagSet.size()) {
+            return true;
+        }
+
+        Optional<String> changedKey = prevEtagSet.stream()
+                .filter(etag -> !newEtagSet.contains(etag)).findFirst();
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigWatch.java]: also method reference

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureConfigPropertySourceLocator.java [hunk 3]
```diff
@@ -74,7 +77,8 @@ private void addPropertySource(CompositePropertySource composite, ConfigStore st
         Collections.reverse(contexts);
         for (String sourceContext : contexts) {
             try {
-                composite.addPropertySource(create(sourceContext, store));
+                List<AzureConfigPropertySource> sourceList = create(sourceContext, store);
+                sourceList.forEach(s -> composite.addPropertySource(s));
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureConfigPropertySourceLocator.java]: method reference here.

Comment (sophiaso) [reply spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureConfigPropertySourceLocator.java]: updated.

**Pull request merged**